pkg/labels/selector.go

```diff
@@ -374,6 +374,10 @@ type Lexer struct {
 	pos int
 }
 
+func NewLexer(s string, pos int) *Lexer {
```

Comment:
> don't you want *unversioned.LabelSelector internally, string externally, and use the same conversion functions ReplicaSet and Deployment use?

Right. But that's easy, we already have functions available for that conversion. `LabelSelectorAsSelector()` returns `Selector` interface from which we can serialize a string. 

I am referring to the other direction here.